### PR TITLE
feat(call): handle Safari device ID rotation and disable AGC

### LINKED ISSUE #1169: Handle Safari device ID rotation with label-based device matching
## Problem

Safari rotates device IDs for privacy (to prevent fingerprinting). This causes a critical UX issue in our video call flow:

1. User selects a device during setup (e.g., external microphone)
2. Device ID is stored in player data (e.g., `micId: "abc123"`)
3. User refreshes the page or browser restarts
4. Safari assigns a **new device ID** to the same physical device
5. The stored device ID no longer matches any available device
6. User ends up with **wrong device** or **no active device**

### Impact

- Users lose their preferred device selection between stages or after refresh
- May end up using built-in microphone instead of headset
- Particularly problematic when users have multiple cameras/microphones
- Creates confusion when audio/video quality differs from setup experience

### Current Behavior

Our device selection during setup stores:
- `cameraId`, `micId`, `speakerId` - Device IDs (Safari rotates these)
- Device IDs don't persist across browser sessions in Safari

## Solution

Implement a **3-tier device matching strategy** that gracefully degrades when IDs don't match:

```
1. ID Match    → Try to find device with exact deviceId match (works on Chrome/Firefox)
2. Label Match → Try to find device with matching human-readable label (Safari workaround)
3. Fallback    → Use first available device of that type (last resort)
```

### Data Model Changes

Store **both ID and label** for each device type during setup:

| Player Key | Description |
|------------|-------------|
| `cameraId` | Preferred camera device ID |
| `cameraLabel` | Preferred camera label (e.g., "FaceTime HD Camera") ✨ NEW |
| `micId` | Preferred microphone device ID |
| `micLabel` | Preferred microphone label (e.g., "MacBook Pro Microphone") ✨ NEW |
| `speakerId` | Preferred speaker device ID |
| `speakerLabel` | Preferred speaker label (e.g., "MacBook Pro Speakers") ✨ NEW |

#### ✅ Completed
- [x] Research and document the Safari ID rotation problem
- [x] Create `deviceAlignment.js` utility with matching logic
- [x] Create comprehensive test suite (`deviceAlignment.test.js`)
- [x] Update README.md with device management documentation
- [x] Implement inline device alignment in `VideoCall.jsx` for camera, mic, and speaker

#### 🚧 In Progress (Current Branch)
- [ ] Store device labels during setup stage
- [ ] Refactor to use the tested utility instead of inline logic

#### 📋 TODO
- [ ] Update `CameraCheck.jsx` to store `cameraLabel`
- [ ] Update `MicCheck.jsx` to store `micLabel`
- [ ] Update `HeadphonesCheck.jsx` to store `speakerLabel`
- [ ] Refactor `VideoCall.jsx` to use `findMatchingDevice()` utility
- [ ] Add Sentry logging for match type (track how often we fall back to labels)

## Files Changed

**New files:**
- `client/src/call/utils/deviceAlignment.js` - Device matching utility
- `client/src/call/utils/deviceAlignment.test.js` - Comprehensive test suite

**Modified files:**
- `client/src/call/VideoCall.jsx` - Implements device alignment (inline, not yet using utility)
- `client/src/call/README.md` - Documents device management strategy
- `client/src/intro-exit/setup/CameraCheck.jsx` - Needs label storage
- `client/src/intro-exit/setup/MicCheck.jsx` - Needs label storage
- `client/src/intro-exit/setup/HeadphonesCheck.jsx` - Needs label storage
- `client/src/call/FixAV.jsx` - Enhanced diagnostic data collection

## Edge Cases Handled

✅ Safari rotates device IDs → Label matching as fallback
✅ Preferred device unavailable → Fall back to first available
✅ `setInputDevicesAsync` throws → Retry with fallback device
✅ Setup not completed (`preferredId === "waiting"`) → Skip alignment
✅ Already using target device → Skip to prevent infinite loops
✅ Empty device arrays → No-op (no devices to align)
✅ Null/undefined values → Graceful handling
✅ Duplicate labels → First match wins

## Suggested Refactoring

The `deviceAlignment.js` utility is **currently not used** by `VideoCall.jsx`. The component has inline logic that duplicates this functionality.

**Before (inline in VideoCall.jsx):**
```javascript
const idMatch = devices?.microphones?.find(
  (mic) => mic.device.deviceId === preferredMicId
);
const labelMatch = !idMatch && preferredMicLabel
  ? devices?.microphones?.find((mic) => mic.device.label === preferredMicLabel)
  : null;
const fallback = !idMatch && !labelMatch ? devices?.microphones?.[0] : null;
```

**After (using utility):**
```javascript
import { findMatchingDevice } from './utils/deviceAlignment';

const result = findMatchingDevice(devices?.microphones, preferredMicId, preferredMicLabel);
if (result) {
  const { device: targetMic, matchType } = result;
  // ... set device
}
```

**Benefits:**
1. Reduce code duplication across camera/mic/speaker alignment
2. Make the logic testable (already tested!)
3. Keep VideoCall.jsx focused on React lifecycle management
4. Enable analytics on match type distribution

## Testing

Run tests with:
```bash
cd client
npm test -- deviceAlignment.test.js
```

The test suite covers:
- ID matching (primary strategy)
- Label matching (Safari workaround)
- Fallback behavior
- Edge cases: empty arrays, null values, duplicate labels, missing fields

## Success Criteria

- [ ] Users retain their device preferences across page refreshes in Safari
- [ ] Graceful degradation when preferred device is unavailable
- [ ] Clear logging of which matching strategy was used
- [ ] Analytics data showing label match frequency (indicates Safari usage)
- [ ] No infinite loops from device switching

## References

- [VideoCall.jsx device alignment logic](../client/src/call/VideoCall.jsx)
- [Device alignment utility](../client/src/call/utils/deviceAlignment.js)
- [Device alignment tests](../client/src/call/utils/deviceAlignment.test.js)
- [Safari privacy fingerprinting protection](https://webkit.org/blog/11545/updates-to-the-storage-access-api/)

## Related Issues

- #1144 - Microphone volume normalization (AGC issue)
- #1163 - Skip loopback check for quiet audio diagnosis

### PR DESCRIPTION
**Pull request by JamesPHoughton**
## Summary

This PR addresses two audio/video issues:
1. **Safari device ID rotation** - Users losing preferred devices after page refresh
2. **AGC experiment** - Testing if disabling browser AGC fixes quiet audio issues

### 1. Safari Device ID Rotation Fix (Closes #1169)

**Problem:** Safari rotates device IDs for privacy, causing users to lose their preferred camera/mic/speaker selections after page refresh.

**Solution:** 3-tier device matching strategy with label-based fallback:
```
1. ID Match    → Exact deviceId match (Chrome/Firefox)
2. Label Match → Match by device label (Safari workaround)  
3. Fallback    → Use first available device (device missing)
```

**Implementation:**
- ✅ Created `deviceAlignment.js` utility with `findMatchingDevice()` function
- ✅ Added comprehensive test suite (21 tests, all passing)
- ✅ Store both `deviceId` and `deviceLabel` during setup for all device types
- ✅ Refactored VideoCall.jsx to use tested utility (eliminated ~60 lines of duplication)
- ✅ Added Sentry monitoring:
  - Breadcrumbs for successful alignments (context)
  - Warning messages when fallback used (device not found)
  - Full diagnostic data: preferred vs actual vs available devices
- ✅ Enhanced FixAV.jsx diagnostics to include label matching status
- ✅ Comprehensive README.md documentation

**Files Changed:**
- `client/src/call/utils/deviceAlignment.js` - New utility function
- `client/src/call/utils/deviceAlignment.test.js` - New test suite (21 tests)
- `client/src/call/VideoCall.jsx` - Refactored to use utility + Sentry logging
- `client/src/intro-exit/setup/CameraCheck.jsx` - Store camera label
- `client/src/intro-exit/setup/MicCheck.jsx` - Store mic label
- `client/src/intro-exit/setup/HeadphonesCheck.jsx` - Store speaker label
- `client/src/call/FixAV.jsx` - Enhanced diagnostics with label matching
- `client/src/call/README.md` - Added comprehensive documentation

### 2. AGC Experiment (Related to #1144)

**Problem:** Users experiencing very quiet audio initially, which normalizes after 2-5 seconds.

**Experiment:** Temporarily disable browser autoGainControl (AGC) to test if it's causing the quiet audio issue. Keeps echo cancellation and noise suppression enabled.

**Status:** Early results suggest disabling AGC has improved the situation - no recent reports of the quiet audio issue in latest batch.

## Testing

```bash
cd client
npm test -- deviceAlignment.test.js
```

All 21 tests passing ✅

## Analytics

Sentry will now track:
- How often Safari users hit label fallback (indicates ID rotation)
- Which device types have more matching issues
- Why users end up with wrong devices
- Correlation with browser/OS versions

**Example Sentry output:**
```
⚠️ Preferred microphone not found, using fallback

Tags:
  deviceType: microphone
  matchType: fallback

Extra:
  preferred: { id: "abc123", label: "Blue Yeti USB Microphone" }
  fallback: { id: "xyz789", label: "MacBook Pro Microphone" }
  availableDevices: [...]
```

## Documentation

See updated [client/src/call/README.md](client/src/call/README.md) for:
- Problem explanation and rationale
- Implementation details
- Sentry monitoring strategy
- Refactoring guidelines for future maintainers

## Related Issues

- Closes #1169 - Safari device ID rotation handling
- Related to #1144 - Microphone volume normalization (AGC experiment)

### CODE REVIEW

File: client/src/intro-exit/setup/MicCheck.jsx [hunk 1]
```diff
@@ -191,15 +191,18 @@ function SelectMicrophone({ onSelected }) {
...
       const selectedMic = devices.microphones.find(
         (mic) => mic.device.deviceId === selectedId
       );
-      logEntry.debug.selectedLabel = selectedMic?.device?.label;
+      const selectedLabel = selectedMic?.device?.label || null;
+      // Store both ID and label - label helps match devices when Safari rotates IDs
+      player.set("micId", selectedId);
+      player.set("micLabel", selectedLabel);
+      logEntry.value = selectedId;
+      logEntry.debug.selectedLabel = selectedLabel;
```
[the remark this answers is not in the document]
Comment (JamesPHoughton) [reply client/src/intro-exit/setup/MicCheck.jsx]: As long as we aren't exporting these values in the scienceData, I think it's alright - I'm not sure how else we'll be able to do it, as we need it to persist across page reloads. I guess we could save it to localstorage with a 3-hour timeout (is that possible?)
